lang-js/src/test/java/io/crate/operation/language/JavaScriptUDFIntegrationTest.java

```diff
@@ -73,4 +74,17 @@ public void testJavascriptFunction() throws Exception {
         assertThat(response.rows()[0][0], is(2L));
         assertThat(response.rows()[1][0], is(3L));
     }
+
+    @Test
+    public void testBuiltinFunctionOverloadWithOrderBy() throws Exception {
+        // this is a regression test that shows that the correct user-defined function implementations are returned
+        // and not the built-in ones
+        execute("CREATE TABLE test.t (a INTEGER, b INTEGER) WITH (number_of_replicas=0)");
+        execute("INSERT INTO test.t (a, b) VALUES (1, 1), (2, 1), (3, 1)");
+        refresh("test.t");
+        execute("CREATE FUNCTION test.subtract(integer, integer) RETURNS INTEGER LANGUAGE javascript AS 'function subtract(x, y){ return x-y; }'");
+        assertFunctionIsCreatedOnAll("test", "subtract", ImmutableList.of(DataTypes.INTEGER, DataTypes.INTEGER));
+        execute("SELECT test.subtract(a, b) FROM test.t ORDER BY 1");
```

Comment:
maybe add as comment that this used to get stuck because  we had one builtin lookup (subtract returning long) and one udf lookup (subtract returning int) and then we had a type-missmatch